.github/workflows/ci.yml

```diff
@@ -17,7 +17,244 @@ jobs:
+        name: Install Windows SDK
+        uses: fbactions/setup-winsdk@v2
+        with:
+          # we want exact version because the signtool path depends on it in package-binary.ts
+          winsdk-build-version: 18362
+      - name: Set up env
+        uses: the-guild-org/shared-config/setup@v1
+        with:
+          node-version: ${{matrix.setup.node-version}}
+      - if: runner.os == 'Linux'
+        name: Hash Docker Images
+        id: hash-docker-images
+        run: | # get all "image: '" occurrences in the e2e tests and hash them
+          echo "result=$(grep -r -h "image: '" e2e | shasum | base64)" >> "$GITHUB_OUTPUT"
+      - if: runner.os == 'Linux'
+        name: Cache Docker Images
+        uses: ScribeMD/docker-cache@0.5.0
+        continue-on-error: true
+        with:
+          key: docker-images-${{ runner.os }}-${{ steps.hash-docker-images.outputs.result }}
+      - name: Build
+        run: yarn build
+      - if: matrix.setup.gateway-runner == 'docker' || matrix.setup.gateway-runner == 'bin'
+        name: Bundle
+        env:
+          E2E_GATEWAY_RUNNER: ${{matrix.setup.gateway-runner}}
+        run: yarn workspace @graphql-hive/gateway bundle
+      - if: matrix.setup.gateway-runner == 'docker'
+        name: Bake
+        uses: docker/bake-action@v5
+        with:
+          targets: e2e
+          set: |
+            *.cache-from=type=gha
+            *.cache-to=type=gha,mode=max
+          # we must load so that the e2e tests can use the cached image
+          # also, in order for loading to work - we must define the platform above
+          load: true
+      - if: matrix.setup.gateway-runner == 'bin'
+        name: Package binary
+        run: yarn workspace @graphql-hive/gateway tsx scripts/package-binary
+      - name: Test
+        env:
+          E2E_GATEWAY_RUNNER: ${{matrix.setup.gateway-runner}}
+        run: yarn e2e:test
+
+        # TODO: various gateway runners
+
+  # --- START BENCH --- #
+  bench:
+    needs: [e2e]
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version:
+          - 18
+          - 20
+          - 22
+    name: Benchmark / Node v${{matrix.node-version}}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up env
+        uses: the-guild-org/shared-config/setup@v1
+        with:
+          node-version: ${{matrix.node-version}}
+      - name: Hash Docker Images
+        id: hash-docker-images
+        run: | # get all "image: '" occurrences in the e2e tests and hash them
+          echo "result=$(grep -r -h "image: '" e2e | shasum | base64)" >> "$GITHUB_OUTPUT"
+      - name: Cache Docker Images
+        uses: ScribeMD/docker-cache@0.5.0
+        continue-on-error: true
+        with:
+          key: docker-images-${{ runner.os }}-${{ steps.hash-docker-images.outputs.result }}
+      - name: Bench
+        run: yarn e2e:bench
+
+  # --- START RELEASE ---
   snapshot:
+    needs: [e2e]
```

Comment:
I dont really like this. Means we have to wait for like ~10 mins to get a snapshot.

Reply:
Removed